app/components/works/show/related_contents_show_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Works
+  module Show
+    # Component for rendering related contents on the work show page
+    class RelatedContentsShowComponent < ApplicationComponent
+      def initialize(work_presenter:)
+        @work_presenter = work_presenter
+        super()
+      end
+
+      attr_reader :work_presenter
+
+      delegate :related_links, to: :work_presenter
+
+      def related_works
+        work_presenter.related_works_attributes.select { |related_work| related_work.to_s.present? }
```

Comment:
If you use `work_presenter.related_works` here, I think you might be able to drop the explicit `to_s` in the block. Like:

```suggestion
        work_presenter.related_works.select(&:present?)
```

Worth a quick try?

Reply:
I messed around with this a bit when I was writing the code, but couldn't get it to work. Just verified with your suggestion:
![image](https://github.com/user-attachments/assets/8bb5fda3-159d-4cde-8d37-1990668b620b)

Reply:
Nevermind, I didn't read your suggestion closely. It works. Fixing.

Reply:
No, I was wrong, it doesn't. `.related_works` returns an array of strings.

Reply:
And what you want here is the instance, got it.